### ISSUE TITLE
integrate(log(0.5-x), (x, 0, 0.5)) wrongly produces imaginary part

### ISSUE THREAD

**Issue by axil** [sympy/sympy#11877]
```
>>> integrate(log(0.5-x), (x, 0, 0.5))
−0.846573590279973+1.0iπ
 
```
 
The `1.0iπ` part is excess. It should be just
```
−0.846573590279973
```

**Comment by jksuom** [comment]
The integral is improper, since the integrand is not defined at the endpoint 0.5, and has to be computed as the limit of integrals over (0, b) as b tends to 0.5. This is carried out in `Expr._eval_interval`. However, the implementation is defective as it will always take the limit from the right. In this example, the antiderivative is branched, and its limit from the left should be used.

**Comment by atemafac** [comment]
hello , i am new here. i will like to work on this issue. please i will need more information on what to do and i will like to be assigned this.

**Comment by jksuom** [comment]
There is a good [introduction to contributing](https://github.com/sympy/sympy/wiki/Introduction-to-contributing) that also contains links to other sources such as an exposition of the practical [workflow](https://github.com/sympy/sympy/wiki/Development-workflow).

**Comment by abhimnc** [comment]
Another newbie here :). I want clarify doubt regarding resolving issue.s Like in this issue I have made changes sympy/sympy/core/expr.py in function _eval_interval(self,x,a,b) but how to check it is correct or not.
I had tried to compiled  integrate(log(0.5-x), (x, 0, 0.5)) but the result is again −0.846573590279973+1.0iπ . I have edited code according to commit made by karan1276 but this also gives the same result. Have I missed something ?

**Comment by karan1276** [comment]
@abhimnc i am already working on this issue. Tho I would not stop your from patching it, it's a good learning experience and maybe you will do it in a better way.

Perhaps you can show me your code?

**Comment by abhimnc** [comment]
I am just a beginner here and want to learn how issues are resolved. @karan1276  can you please tell me how you verified yourself after making changes. Have you checked the output of `integrate(log(0.5-x), (x, 0, 0.5))` by editing ? I have tried doing this but it doesn't make any change in result. If this is the way to check  then may be there will be no improvement ny making change in `_eval_interval(self,x,a,b).` We may have to look in integral class for this.
But first tell me the process of verification.

**Comment by karan1276** [comment]
@abhimnc yes, i just run the function and check the output. in my code i get the answer, if you do everything right you should get the answer too. Maybe you are missing something.

**Comment by chinmay0301** [comment]
Hi, is anyone still working on this bug? Another newbie here, and wanted to try fixing it

**Comment by sarthak268** [comment]
Please assign me this issue

**Comment by rahul-prasad-cse** [comment]
is this issue still open?

**Comment by jtoka** [comment]
Added fix. @axil @jksuom Are you able to validate?

**Comment by jksuom** [comment]
Fixed by #11889.
